src/utils/transform.ts

```diff
@@ -117,15 +118,14 @@ export default function transform(
 	let promise = Promise.resolve(source.code);
 	let transformDependencies: string[];
 
-	plugins.forEach(plugin => {
-		if (!plugin.transform) return;
-
+	plugins.filter(plugin => 'transform' in plugin).forEach(plugin => {
+		const transform = cacheTransformHook(graph.cachedHooks, plugin);
 		promise = promise.then(previous => {
 			const { assets, emitAsset } = createTransformEmitAsset();
 			return Promise.resolve()
 				.then(() => {
 					const context = createPluginTransformContext(graph, plugin, id, previous, emitAsset);
-					return plugin.transform.call(context, previous, id);
+					return transform.call(context, previous, id);
```

Comment:
I'd be interested to know if you found this makes any performance difference, as we actually already do cache the transform hook (this is the only thing we cache currently). There is a check [here](https://github.com/rollup/rollup/blob/master/src/Graph.ts#L690) that reuses the transformed result when the input source is matched. Ideally this should be cacheKey based though as transform options changing would currently not invalidate this.

The current cache also deals with the `emitAsset` problem by noting which assets are emitted by the transform hook and then automatically re-emitting them when using the cached transform. I'm hoping to get a PR together tomorrow which can do a similar thing for the load function to hopefully unblock this caching direction.

Reply:
I do remember seeing this - and attempting to get it persisting to disk, but for some reason couldn't. I can't remember the exact reasons why. I'm happy to move this from this PR if I can figure out how to get the module cache working 😃

Reply:
Would be very interesting to know the stumbling block as you likely wouldn't be the first on that either :)